components/script/script_thread.rs

```diff
@@ -1172,9 +1172,13 @@ impl ScriptThread {
                 );
             },
         };
-        let _ = window
-            .layout_chan()
-            .send(Msg::RegisterPaint(name, properties, painter));
+
+        match window.layout_chan() {
+            Some(chan) => chan
+                .send(Msg::RegisterPaint(name, properties, painter))
+                .unwrap(),
+            None => warn!("Layout channel unavailable"),
+        }
```

Comment:
Not sure why the `Result` was ignored here and not unwrapped like everywhere else, I've changed it to match with other uses of this channel